### DISCUSSION TITLE
PR Release notes with self-hosted bot and on-premise Azure DevOps 2020 Server packages

### DISCUSSION THREAD

**Discussion by japj** [renovatebot/renovate#8665] [Suggest an Idea]
Hello,

I have recently started using a self-hosted renovatebot with an Azure DevOps 2020 Server and I am really happy with the results.
So far I've mostly run it on a project that consumes public npm and nuget packages and I noticed a fair bit of difference in the available 'release notes' information in the PRs.

Since we are also building our own (private) npm and nuget packages I am trying to understand better how to make sure we get the "best" release notes experience for PRs related to those. I am having some trouble understanding everything that renovate is trying to do related to gathering of changelog/release notes.

Is there some information on what package authors (for nuget/npm) need to do to ensure the best PR experience with Renovate?

Also: nuspec has a releaseNotes field (sometimes containing the url to the CHANGELOG.md file, or actual relevant changes), perhaps that could also be of use? (as far as I can tell it is currently ignored in renovate?)

We also use https://keepachangelog.com/en/1.0.0/ and I was wondering if there is a way to use that to extract release notes per package version information.

**Comment by rarkins** [comment]
The package registry must include the source URL in its metadata. Then, we can fetch the changelog from the source repository if it's GitHub or Gitlab (so far). Other platforms can be added but needs code
